proposed/2022/dotnet-nuget-why-proposal.md

```diff
@@ -0,0 +1,140 @@
+
+![](../../meta/resources/TransitiveDependencies/SolutionView.png)
+
+The dotnet CLI does not provide the insight into “why” a transitive dependency is listed, although it does list everything that has been resolved.
+
+![](../../meta/resources/TransitiveDependencies/DotNetCLI.png)
+
+We could have considered a minor tweak to the existing experience of “dotnet list package --include-transitive” to provide the user with a sense of where the package came from. This could be a new column next to “Resolved” which says “Transitively Referenced” or “Referenced” and has a list of the top-level packages that requested the dependency. Tracking issue can be found here: https://github.com/NuGet/Home/issues/11625. Unfortunately, this solution doesn't provide an easy way to identify the dependency graph `(starting from project -> top-level package -> transitive-dependency (1) -> ..-> transitive dependency (n))` for a particular package.
+
+![](../../meta/resources/TransitiveDependencies/TransitiveDotNetCLI.png)
+
+Within Visual Studio, there will be a new panel within the Visual Studio Package Manager UI named “Transitive Packages” in which all transitive packages will be displayed to the user. For the sake of not confusing the user, there will additionally be a header titled “Top-level Packages” where the current experience is today for top-level packages.
+
+When a user highlights a transitive package, they will see a pop-up that displays how the transitive dependency originated & what top-level package(s) are bringing it in.
+
+![](../../meta/resources/TransitiveDependencies/TransitiveVSPMUI.png)
+
+Sadly, we don’t have any further CLI or visualizing tool today to answer `Running 'dotnet restore' on a (large) .sln file restores a certain problematic package. How do I know which project(s) in the solution is causing that package to be restored?` question.
+
+## Goals
+The main goal is to add `dotnet nuget why` command to .NET CLI to help customers understand the dependency graph of a given package for PackageReference projects.
+
+Projects that use `packages.config` format are out of scope for this effort because the XML file contains a flat list of both direct and transitive package dependencies. Customers cannot visualize the dependency graph from the file itself but it would be easy to update the package version of transitive depedency incase if there is any security vulnerability in the current version.
+
+## Customers
+We are building this for .NET users to be successful with understanding dependency graph for a given package.
+
+## Solution
```

Comment:
The more I read it, I don't think we're using the correct template for this proposal: https://github.com/NuGet/Home/blob/dev/meta/template.md.

Reply:
Based on other comments, I agree. I can change the format to more closely resemble the template you linked.